feat(python): add get_flag method

KozyarValeriy opened the pull request:
In this PR, I have added support for the `get_flag` API. I have also fixed several issues that violated PEP8 and added new rules for Ruff. 

We use this API to get a specific flag, because it is expensive to request all flags when there are many of them. In our case, we sometimes need to get a single specific flag and don't want to get a list and sift through them all.

If there are actually no plans to implement this feature in the SDK, please let me know so I will remove these changes from the PR and keep the code style edits. If you do not think this is necessary, please let me know, and I will close this PR.

markphelps reviewed (approved):
this looks great to me @KozyarValeriy !! thank you for this!!

I'll update the other issues for the other languages (ie: https://github.com/flipt-io/flipt-server-sdks/issues/131) to include `GetFlag` in addition to `ListFlags`

KozyarValeriy commented:
I'm very sorry, but it has just occurred to me that this PR have broken backward compatibility with version 1.2.0 due to renaming params. I replaced camelCase with snake_case for parameters in Flag, ListParameters and PaginatedResponse models. The use of camelCase for Python is not recommended ([pep8](https://peps.python.org/pep-0008/)), so I have replaced it throughout the code. I needed to add BREAKING_CHANGES.md, but I had forgotten about it.

Also I have added some new rules for ruff so that in the future, the linter will not pass with incorrect parameter names.

I'm really sorry for any trouble this might have caused.

markphelps commented:
All good, no problem @KozyarValeriy !
